src/modules/task/task.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, Query } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Task } from './task.entity';
+import { FindOperator, Repository } from 'typeorm';
+import { VTask, VTaskOpt } from './task.validation';
+import { ILike } from '../../utils/ilike';
+import { Pagination } from '../../validators/pagination';
+
+@Injectable()
+export class TaskService {
+  constructor(@InjectRepository(Task) private taskRepo: Repository<Task>) {}
+
+  async findById(id: number) {
+    return await this.taskRepo.findOne(id);
+  }
+
+  async listAll(
+    { skip, take }: Pagination,
+    search: string,
+    userId: number,
+    projectId: number,
+  ) {
+    let where = [
+      { title: ILike(`%${search}%`) },
+      { description: ILike(`%${search}%`) },
+      { state: ILike(`%${search}%`) },
+    ];
+    let filter = {};
+    if (userId) {
+      filter = { ...filter, userId: new FindOperator('equal', userId) };
+    }
+    if (projectId) {
+      filter = { ...filter, projectId: new FindOperator('equal', projectId) };
+    }
```

Comment:
To mi deluje malo hacky, če imaš boljši predlog LMK :)

Reply:
Logika enaka tvoji, samo na krajše:

    const filter: any = {};
    if (userId) {
      filter.userId = userId;
    }
    if (projectId) {
      filter.projectId = projectId;
    }

FindOperator, ne rabiš nikoli direktno klicat. Za equals ga pa sploh ne rabiš.
Za ostale primerjave pa obstajajo tile: [advanced-options](https://github.com/typeorm/typeorm/blob/master/docs/find-options.md#advanced-options).
Je pa res da ILike ni obstajal, pa sem ga sam dodal v projekt (file /utils/ilike.ts).